firefox-ios/Client/Application/SceneDelegate.swift

```diff
@@ -162,7 +158,21 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
             guard let urlString = tab["url"] as? String,
                   let url = URL(string: urlString),
                   let route = routeBuilder.makeRoute(url: url) else { continue }
-            sceneCoordinator?.findAndHandle(route: route)
+            handle(route: route)
+        }
+    }
+
+    private func handle(route: Route) {
+        guard let sceneCoordinator = sceneCoordinator else {
+            logger.log("Scene coordinator should exist", level: .fatal, category: .coordinator)
+            return
+        }
+
+        logger.log("Scene coordinator will handle a route", level: .info, category: .coordinator)
+        sessionManager.launchSessionProvider.openedFromExternalSource = true
+
+        AppEventQueue.wait(for: [.startupFlowComplete, .tabRestoration(sceneCoordinator.windowUUID)]) {
+            sceneCoordinator.findAndHandle(route: route)
         }
```

Comment:
I think there is a decent chance this will catch some of the issues we are seeing. We should try to get it into a beta to test it out as soon as we can.

Reply:
This is a fix I explored previously while investigating 7799 but unfortunately this code is never called to begin with (at least for me). But maybe my data is wrong or something was unusual about how the bug repro'd for me (and I've still only hit it a few times).

Reply:
I haven't been able to reproduce the problem, but my thinking is just... worst case it will be the same, but we'll have more logs at least 😅